cleanlab/dataset.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Dataset Module
+Supports dataset-level and class-level automated quality, including finding which classes should
+be merged with other classes in your dataset, which classes should be removed,
+and which classes have the best label accuracy, and which have the worst labels.
+"""
```

Comment:
It could be nice if all the functions in this file support optional arg:
```class_names = None```
where user can provide mapping between string names for the classes and int representations.
Eg. `class_names` supported inputs could be dict and list:
```
{0: 'classA', 1: 'classB',...}
```
and 
```
['classA', 'classB']
```

Then if `class_names is not None`, all print statements / outputs are presented in terms of class names instead of ints.

This can also be for future PR.

Reply:
it actually even simpler! just needs to be a list with the class names in order (because we have guarantees on the format of `labels`).

Reply:
I was suggesting either list or dict input formats could be supported for this function (i.e. both types work).  When I use cleanlab, I personally prefer to track the mapping between class-names and ints as a dict in my own code before running cleanlab commands, because dict is much more interpretable when I print it.

Reply:
Went ahead and implemented this for 2.0 🎉